docusaurus/docs/create-a-plugin/extend-a-plugin/add-resource-handler.md

```diff
@@ -0,0 +1,169 @@
+
+To add a resource handler to your backend plugin, you need to implement the `backend.CallResourceHandler` interface for your data source struct.
+
+```go
+func (d *MyDatasource) CallResource(ctx context.Context, req *backend.CallResourceRequest, sender backend.CallResourceResponseSender) error {
+    return sender.Send(&backend.CallResourceResponse{
+        Status: http.StatusOK,
+        Body: []byte("Hello, world!"),
+    })
+}
+```
+
+You can then access your resources through the following endpoint: `http://<grafana instance host:port>/api/datasources/uid/<DATASOURCE_IUD>/resources`
+
+In this example code, `DATASOURCE_UID` is the data source unique identifier (UID) that uniquely identifies your data source.
+
+:::tip
+
+To verify the data source UID, you can enter `window.grafanaBootData.settings.datasources` in the Browser Developer Tools Console, to list all the configured data sources in your Grafana instance.
+
+:::
+
+## Add support for multiple routes
+
+To support multiple routes in your data source plugin, you can use a switch with the `req.Path`:
+
+```go
+func (d *MyDatasource) CallResource(ctx context.Context, req *backend.CallResourceRequest, sender backend.CallResourceResponseSender) error {
+	switch req.Path {
+	case "namespaces":
+		return sender.Send(&backend.CallResourceResponse{
+			Status: http.StatusOK,
+			Body:   []byte(`{ namespaces: ["ns-1", "ns-2"] }`),
+		})
+	case "projects":
+		return sender.Send(&backend.CallResourceResponse{
+			Status: http.StatusOK,
+			Body:   []byte(`{ projects: ["project-1", "project-2"] }`),
+		})
+	default:
+		return sender.Send(&backend.CallResourceResponse{
+			Status: http.StatusNotFound,
+		})
+	}
+}
+```
+
+You can also query your resources using the `getResource` and `postResource` helpers from the `DataSourceWithBackend` class.
+
+For example, in your query editor component, you can access the data source instance from the `props` object:
+
+```
+const namespaces = await props.datasource.getResource('namespaces');
+props.datasource.postResource('device', { state: "on" });
+```
+
+## Advanced use cases
+
+If you have some more advanced use cases or want to use a more Go-agnostic approach for handling resources, you can use the regular [`http.Handler`](https://pkg.go.dev/net/http#Handler). You can do so by using a package provided by the [Grafana Plugin SDK for Go](../../introduction/grafana-plugin-sdk-for-go.md) named [`httpadapter`](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend/resource/httpadapter). This package provides support for handling resource calls using an [`http.Handler`](https://pkg.go.dev/net/http#Handler).
+
+What’s interesting with using [`http.Handler`](https://pkg.go.dev/net/http#Handler) is that you can use it with Go’s builtin router functionality called [`ServeMux`](https://pkg.go.dev/net/http#ServeMux) or use your preferred HTTP router library (for example, [`gorilla/mux`](https://github.com/gorilla/mux)).
```

Comment:
```suggestion
What’s interesting with using [`http.Handler`](https://pkg.go.dev/net/http#Handler) is that you can use it with Go’s built-in router functionality called [`ServeMux`](https://pkg.go.dev/net/http#ServeMux) or use your preferred HTTP router library (for example, [`gorilla/mux`](https://github.com/gorilla/mux)).
```

Reply:
"What's interesting" seems more conversational than usual and we don't really explain why its useful to use ServeMux or other libraries. Not a blocker, just a suggestion
```suggestion
Using [`http.Handler`](https://pkg.go.dev/net/http#Handler) allows you to also use Go’s built-in router functionality called [`ServeMux`](https://pkg.go.dev/net/http#ServeMux) or your preferred HTTP router library (for example, [`gorilla/mux`](https://github.com/gorilla/mux)).
```